src/Module.zig

```diff
@@ -705,6 +705,15 @@ pub const Decl = struct {
         var buffer = std.ArrayList(u8).init(mod.gpa);
         defer buffer.deinit();
         try decl.renderFullyQualifiedName(mod, buffer.writer());
+
+        // Sanitize the name for nvptx which is more restrictive.
+        if (mod.comp.bin_file.options.target.cpu.arch.isNvptx()) {
+            for (buffer.items) |*byte| switch (byte.*) {
+                '{', '}', '*', '[', ']', '(', ')', ',', ' ', '\'' => byte.* = '_',
+                else => {},
+            };
+        }
```

Comment:
Is this not gonna cause linker errors when two symbols map to the same value? Say functions `@"A()"` and `@"A{}"`.

Reply:
yes, it's true. I don't know how robust this should be. I could sanitize and append some hash. Would that be better ?